### DISCUSSION TITLE
Conflicting info in wiki vs sample script

### DISCUSSION THREAD

**Discussion by GabeShack** [grahampugh/erase-install#413] [General]
Im seeing some conflicting info regarding using the Jamf-Pro script and parameters:

The Wiki states:
"Just to reiterate, any parameter can be supplied in any order in the policy's Script Parameters section, and only **one** parameter can be supplied into each Parameter field."

However the sample script says:
"You can use Jamf Pro parameters 1-10 to supply arguments to erase-install,
and you can supply **multiple** arguments in one Jamf Pro parameter."

Wondering which is accurate?

**Comment by GabeShack** [comment]
![Screenshot 2023-07-06 at 10 19 05 AM](https://github.com/grahampugh/erase-install/assets/53574780/b10ad942-b70e-4d48-898b-67e66aee3d81)

**Comment by GabeShack** [comment]
My reason for this question being I have more commands than will fit in parameters 1-10

**Comment by grahampugh** [comment]
Hi @GabeShack. Both are accurate. If you upload erase-install.sh directly into Jamf, you have to supply each parameter separately. Equals signs are required for those parameters that require a value, e.g. `--rebootdelay=120`.

However, if you deploy erase-install as a package, you can upload the `erase-install-launcher.sh` script to Jamf. _This_ script allows multiple parameters per Jamf Parameter Value field.

Alternatively you can also deploy erase-install as a package and just use the "Run Command" field to provide as many parameters as you like, e.g. `/Library/Management/erase-install/erase-install.sh --reinstall --update --confirm --rebootdelay 120 --preinstall-command "rm -rf /Applications/Google\ Chrome.app" --os 12` etc etc.

**Reply by grahampugh** [reply]
I'm interested to know how you got to more than 6 parameters though! I see a lot of people using unnecessary or obsolete parameters.

**Reply by GabeShack** [reply]
![Screenshot 2023-07-06 at 10 32 50 AM](https://github.com/grahampugh/erase-install/assets/53574780/b69629cb-0839-4593-a3cc-9e2558a4f49f)

Tell me if any of these look obsolete to you?

**Reply by GabeShack** [reply]
Was playing with the secure boot level command as well...but i could probably remove that.

**Reply by grahampugh** [reply]
I would say these are unnecessary:
- `--version 13.4.1` -> since you are calling the latest version, you don't need this.
- `--postinstall-command "jamf recon"` -> this most probably doesn't do anything useful that you think it will.

**Reply by GabeShack** [reply]
Well I like to keep 13.4.1 hard coded so if a new version of the OS comes out, it doesn't use it until we have tested it.  I assume that the post install command just runs after the startosinstall, so it still wont tell jamf about the new os install....although this is the erase and not the upgrade so yes i can lose that one for sure.  I think im going to add a preinstall trigger to place the files i want into the extras folder.

**Comment by GabeShack** [comment]
Thank you.  Im toying with the idea of using the extras folder for signed package installers which would save us another script i call during enrollment complete, which pushes the base apps of our image.  Doing this before would save some startup and login time so just starting to play around with my next years image.

**Comment by GabeShack** [comment]
Would you be able to tell me the error message that gets displayed when you try to use an unsigned package in the extras folder?

**Reply by grahampugh** [reply]
Good question. I'm not the expert there, but I don't think you get an error, it just doesn't install. You could check in install.log I guess. You could also ask in the #eraseinstall Slack channel in case some other people have experience with that - I think more people hang out there than read these discussions.
